### DISCUSSION TITLE
Show hostname (or device name), ip address and tasmota version on sending teleperiod

### DISCUSSION THREAD

**Discussion by helmar74** [arendst/Tasmota#21424] [Ideas]
I'm currently thinking of building up a tasmota inventory in my smarthome.
Therefore it would be great get hostname (or device name), the current ip address an tasmota version with sending teleperiod.
With this information it could be identified which device hast which ip address. (MAybe the addresses changes due to DHCP refresh)
Maybe it would make sense to make this feature configurable with a setoptionXXX ?
Maybe it would be enough to create a setoption (when enabled) which sends out status 2 and status 5 on teleperiod?

**Comment by sfromis** [comment]
The IP address does not change, unless possibly if Tasmota is rebooting. And the boot time messages published via MQTT already include the info you ask for. By default, the discovery payloads are even retained, meaning that you can get the whole inventory immediately after subscribing, instead of waiting for telemetry messages to trickle in.

In addition to the discovery messages, the Info1/2/3 payloads at boot also has the info you ask for, and can be made retained using the `InfoRetain` command.

**Reply by scargill** [reply]
Ok as a follow up from that - is there any way (without using Berry) to get tasmota to display it's IP sddress.. could it get it from the MQTT messages and somehow turn that int displaytext info?

**Reply by sfromis** [reply]
You can pick it up from the already mentioned boot time Info2 payload with a rule construct like
`on Info2#IPAddress do var1 %value% endon`
Or from the telemetry STATE messages, by default every 5 minutes, this way:
`on tele-IPAddress do var1 %value% endon`

Of course, you could also have a `DisplayText` command there, instead of setting `var1`

Beware that ESP32 devices with Ethernet can react to two IP addresses, both Ethernet and Wifi.
